app/src/test/java/org/simple/clinic/user/UserSessionTest.kt

```diff
@@ -868,4 +882,24 @@ class UserSessionTest {
   private fun `params for logout clear database failures`(): List<Any> {
     return listOf(NullPointerException(), RuntimeException())
   }
+
+  @Test
+  @Parameters(method = "params for logout report pending sync records failures")
+  fun `when reporting pending records fails, the logout must fail`(cause: Throwable) {
```

Comment:
I think that reporting analytics should not affect logout per se.
Reporting should be a side effect and not part of the flow.
We should make this a fire and forget call.

Reply:
Good point. I'll make this change.

Reply:
Done.